### PR TITLE
Kubernetes SD example: separate out cluster level components & services

### PR DESCRIPTION
**Pull request by jimmidyson**
/cc @keegancsmith

### CODE REVIEW

File: documentation/examples/prometheus-kubernetes.yml [hunk 1]
```diff
@@ -1,22 +1,57 @@
...
+# and services to allow each to use different authentication config.
+#
+# Kubernetes labels will be added as Prometheus labels on metrics.
 
+# Scrape config for cluster components.
 scrape_configs:
-- job_name: 'kubernetes'
+- job_name: 'kubernetes-cluster'
+
+  tls_config:
```
Comment (brian-brazil) [inline documentation/examples/prometheus-kubernetes.yml]: You're using auth in two different ways, comments as to which is which would help users.

File: documentation/examples/prometheus-kubernetes.yml [hunk 2]
```diff
@@ -1,22 +1,61 @@
-# A scrape configuration for running prometheus in cluster on kubernetes. It
-# will create endpoints for node and master roles, as well as any service
-# which is annotated with `prometheus_io_scrape=true`
+# A scrape configuration for running prometheus in cluster on kubernetes.
```
Comment (juliusv) [inline documentation/examples/prometheus-kubernetes.yml]: Hmm, maybe "in cluster on Kubernetes" -> "on a Kubernetes cluster" or "in the cluster on Kubernetes"?

Capitalize proper nouns Prometheus and Kubernetes please.

File: documentation/examples/prometheus-kubernetes.yml [hunk 3]
```diff
@@ -1,22 +1,61 @@
...
+# This uses separate scrape configs for cluster components (i.e. master, node)
+# and services to allow each to use different authentication config.
```
Comment (juliusv) [inline documentation/examples/prometheus-kubernetes.yml]: s/config/configs/

File: documentation/examples/prometheus-kubernetes.yml [hunk 4]
```diff
@@ -1,22 +1,61 @@
...
+#
+# Kubernetes labels will be added as Prometheus labels on metrics.
```
Comment (juliusv) [inline documentation/examples/prometheus-kubernetes.yml]: Maybe append "...via the `label_map` relabeling action" to clarify that this doesn't happen by default.

**Comment by juliusv** [comment]
:+1: awesome otherwise, thanks a lot!

**Comment by juliusv** [comment]
Thanks for the additions!

**Pull request merged**